### ISSUE TITLE
Create naming pattern for getting started content

### ISSUE THREAD

**Issue by szabosteve** [elastic/docs#1096]
The aim of this ticket is to establish a naming pattern for GS & tutorial content that takes SEO into account.

Resources:
* [Getting started vs Quick start content](https://docs.google.com/document/d/1Y9XhYV1qYuKHSw-zgIZxSkjPKqUoYdw30HkLEdfcsLY/edit#heading=h.r1khm0bcatz9)
* [Getting started follow-up meeting notes](https://docs.google.com/document/d/1Oh1YNjfiHzyrluFKB8KVzyhlfIkGwL-q6CSZ6JwkDSs/edit#)
* [Getting started with Elasticsearch](https://www.elastic.co/guide/en/elasticsearch/reference/master/getting-started.html)
* [Writing style guide](https://wiki.elastic.co/display/DOC/Writing+style+guide)

**Comment by alaudazzi** [comment]
Links to SEO related issues in other GH repos:

- [#29108](https://github.com/elastic/cloud/issues/29108)
- #54

**Comment by szabosteve** [comment]
# Getting started

## Working definition

Getting started content is targeted more towards people who want to set things up and understand how things work. For novice users who want to learn.
How to install, configure, and run a basic workflow.


## Page title (H1) and file ID (URL pattern)

Based on the DEVO GS workgroup discussions, GS content title should refer to the topic that it is written for. To name a page as getting started, because the menu structure refers to the broader topic anyway is not an ideal solution from an SEO perspective and it puts an unnecessary cognitive load to the reader. Keep in mind that the topic title is also used for the navigation title.

**Suggested title pattern**:

Getting started with {product/feature}

Example:

`Getting started with Elasticsearch
`

To achieve better SEO, the URL pattern should reflect the page title (H1). You can control the URL pattern by giving the header of the asciidoc file an ID. The HTML page URL will be generated from this ID. Omit connectives, prepositions, etc. and use dashes to separate the words in the ID. Ideally, corresponding topics (e.g.: GS) for different products would have the same name so that you can change the product name in the URL to jump between them. Same applies to anchors (see below).

**Suggested ID**:

[[getting-started-{product/solution}]]

Example:

```
[[getting-started-elasticsearch]]
= Getting started with Elasticsearch
```


## H2-H3 and anchor text

From H2 and beyond, use action verbs in titles instead of gerunds as the Writing style guide suggests it.

Example:

== Enable Elasticsearch security features


Define anchors for each heading, the anchor text is used for cross-reference links. Don't forget, if the heading begins a new page, the anchor text is used for the page name in the generated HTML (as described above). Anchors should reflect the heading title for the sake of SEO, use dashes to separate words in the ID, omit connectives and prepositions, etc. 

General guidelines (from the Writing style guide):

* Anchors should reflect the keywords from the heading and topic.
* Avoid abbreviations. Abbreviations can hurt when it comes to SEO. For example, if a page is about Elasticsearch security, the anchor is set to [[es-sec]], and the name of the page is es-sec.html, it’s not going to perform as well as it would if the page name was elasticsearch-security.html.

Example:

[[enable-elasticsearch-security-features]]
== Enable Elasticsearch security features


# Tutorial

## Working definition

Tutorials are for users who want to work through a scenario to understand the product/service/solution and how they could use it. Step-by-step procedure built around a particular data set and use case. A tutorial seeks to teach by example and supply the information to complete a certain task.


## Page title (H1) and file ID (URL pattern)

The principle is the same as above.

**Suggested title pattern**:

{product/solution/feature} tutorial

OR

Tutorial: {product/solution/feature}

Example:

`Machine learning tutorial
`
OR

`Tutorial: Encrypting communications
`

**Suggested ID**:

[[{product/solution}-tutorial]]

OR

[[tutorial-{product/solution}]]


Example:

```
[[machine-learning-tutorial]]
= Machine learning tutorial
```

OR

```
[[tutorial-encrypting-communications]]
= Tutorial: Encrypting communications
```

## H2-H3 and anchor text

The principle is the same as above.

**Comment by alaudazzi** [comment]
@szabosteve Awesome recap!

We may want to simplify the Tutorial definition with something like that:

> A tutorial walks you through a scenario to let you quickly learn about a product/service/solution. It provides a step-by-step procedure to complete a task for a particular use case.

Also, I find this structure easier to read:

> ### Suggested title pattern:
> {product/solution/feature} tutorial
> Machine learning tutorial
> 
> OR
> 
> Tutorial: {product/solution/feature}
> Tutorial: Encrypting communications
> 
> ### Suggested ID:
> [[{product/solution}-tutorial]]
> [[machine-learning-tutorial]]
> Ex. = Machine learning tutorial
> 
> OR
> 
> [[tutorial-{product/solution}]]
> [[tutorial-encrypting-communications]]
> Ex. = Tutorial: Encrypting communications

Food for thoughts: Should we also add a note about prerequisites? Less for a GS, more for a tutorial. Or maybe this is more related to the doc internal structure.

**Comment by szabosteve** [comment]
# Conceptual overview

[[ml-<feaure>]]
= <Feature_name>

////
The conceptual overview summarizes the main idea and machinery behind the feature. The aim of this high-level summary is to inform the user about the main functions, to familiarize the domain language, basic terms, and to provide some context about the possible fields of usage (short examples, use cases).

The overview section may contain further sub-sections: mind the appropriate heading levels.
////


# How to / Configuration

////
This section shows how to use / configure the feature or contains information related to the feature that couldn’t be the part of the conceptual overview for some reason.
////

# Examples

[[ml-<feaure>-examples]]
== <feaure> examples
++++
<titleabbrev>Examples</titleabbrev>
++++

////
Examples are detailed scenarios of using the feature to solve a specific problem or to cover a use case. Their aim is to show the capabilities of the feature and give the user a sense of its mechanism.
////


# Limitations

[[ml-<feaure>-limitations]]
== <Feature_name> limitations
++++
<titleabbrev>Limitations</titleabbrev>
++++

//// List and description of known issues, limitations of the feature.

The following limitations and known problems apply to the <version number> release of the <name of the feature>:

[float]
[[<feature>-<reference_to_the_limitation>-limitations]]
=== <Limitation_name> limitation

//// Short description of the limitation.


# API quick reference
[[ml-<feature>-apis]]
== API quick reference

All <feature> endpoints have the following base:

[source,js]
----
/_ml/<base_path>
----
// NOTCONSOLE

//// List of links of the API endpoints


# Troubleshooting

[[ml-<feature>-troubleshooting]]
== Troubleshooting <feature>
++++
<titleabbrev>Troubleshooting</titleabbrev>
++++

//// High-level overview of common problems and answers to frequently asked questions.

Use the information in this section to troubleshoot common problems and find
answers for frequently asked questions.

//// List of topics

**Comment by szabosteve** [comment]
@alaudazzi @debadair @Titch990 

Thank you for your feedback! I change the definition of tutorials along with the lines of Arianna's suggestion and mentioned prerequisites as one of the factors that can make the distinction between the two content types.

> we should be framing things in terms of user goals, not the product architecture

I agree, so I amended the proposal above accordingly. It is hard to change the product-specific perspective that a technical writer may inherently have, so I think it is important to highlight this principle.

I removed the less popular options, so we have a clearer idea now.

**Comment by Titch990** [comment]
Brilliant Steve!

I agree with @alaudazzi  about the tutorial titles, and I see you have already modified that. 

I have a pet peeve, and I wonder whether it's worth mentioning here in our guidelines for Getting Started topics?

One of the things that helps send readers down rabbit holes is _indiscriminate linking_. I think it's particularly important to avoid unnecessary links in getting started topics. Readers can get lost in ways we can't imagine, and every link we provide takes the reader away from the getting started topic, and increases the number of potential paths they can take, depending on whether they click the link, and what they do next if they do. In a getting started topic, to ensure reader success, I think we need to control the reader's path through the content as far as possible. 

Some links will be necessary, but I think every one needs to earn its worth. Necessary links could be:

- a link to a necessary concept we are assuming the reader already understands (for example, if they are installing something that needs a basic licence, we could provide a link to a topic that explains what a basic licence is, if they don't know)

- a link to something they need to have done before starting the instructions we provide (for example, checking they are using a supported OS and have the required resources for the installation)

- a link to some next steps they may like to take after they have successfully completed our instructions 

In every case, I think:

- the context surrounding the link should make it clear what information the link provides, and give the reader enough information to decide whether they need to click it

- the linked content must deliver exactly what the reader needs at that point, no more, no less

- if the reader needs to take some actions as a result of the link (for example, signing up for a free trial), the getting started content immediately after the link should meet the reader at the point they should have now reached, and if necessary, help them confirm they have been successful.

These are all standard guidelines for appropriate use of links, but I think they are particularly important when we are showcasing our product and helping readers get started.

**Comment by Titch990** [comment]
@debadair @szabosteve @alaudazzi I have a question on which I'd like your opinion.

In my content, I have several topics currently called "Getting started with xxx" (or occasionally, "Installing xxx"). Because Observability is a product (several products, actually), built on ElasticSearch and Kibana, all these topics follow a similar format. In order to use the product, you have to install ElasticSearch and Kibana, or already have an installation, (or get a cloud trial, or already have a cloud installation). Then you have to install and configure some other product-specific stuff. Then you can start using the product.

So these topics don't quite meet either definition above. They aren't just a low friction way of doing something interesting, although they might be for some readers, nor are they just a step-by-step procedure to complete a task, although again, they might be for some readers. They are fundamental installation and configuration instructions, without which the product or component can't be used. Additionally, the reader may or may not be the same person who is interested in learning how to use the product.

Currently, many of these topics have "Getting started" in the title. I suspect we should be moving towards something more specific, like "Installing and configuring xxx". 

What do you think?  I have a few topics like this in my area, which I'm working around at present, and I'm sure there are many more. 

Making a conscious decision not to use "Getting started" in the title of these topics would help stop these topics being returned incorrectly as hits for our genuine "Getting started" content too. Although if we want to change the existing topic file names to match for improved SEO, we will probably need to start adding redirects?

**Comment by alaudazzi** [comment]
+1 to rename these topics towards something more specific, like "Installing and configuring xxx".
Can you also provide some examples of the topics currently called "Getting started with xxx" ?

**Comment by szabosteve** [comment]
> +1 to rename these topics towards something more specific, like "Installing and configuring xxx".
> Can you also provide some examples of the topics currently called "Getting started with xxx" ?

If they don't meet with the GS definition, then I also think they should be renamed.
For this, it is necessary to collect the respective pages in a separate issue.

@Titch990 I absolutely agree with you about the linking guidelines that you mentioned before. I add your suggestion to the DEVO working document that I linked in the first comment.

**Comment by Titch990** [comment]
An example topic is https://www.elastic.co/guide/en/infrastructure/guide/current/install-infrastructure-monitoring.html (topic title is "Getting started with infrastructure monitoring"). This is really "installing and configuring". It's stuff you need to do before you can see the results of monitoring anything.

The Observability docs have several other similar (not always consistent, and not always quite correct) topics. I'm about to raise a general ticket to fix these and make them consistent and correct. The installation instructions aren't always clear and don't always flow very well either, but that's a separate issue from the overall title.

**Comment by gchaps** [comment]
These are great guidelines, for getting started, tutorials, and linking.  Will you be adding this information to the [Docs writing style guide](https://wiki.elastic.co/display/DOC/Writing+style+guide)?

**Comment by Titch990** [comment]
@gchaps. I think some of what we have is notes rather than guidelines ready for the Style Guide, although in time we all hope to have something more concrete.

You may have noticed that I have a bit of a thing about good quality links, so feel free to grab that bit now if you like it. Or I can condense it a bit and add it myself if you prefer?

**Comment by gchaps** [comment]
@Titch990 I would prefer if you add the information about links as you are the expert in this area.

**Comment by gchaps** [comment]
The Kibana docs have short examples with titles such as "Example: Register a file system repository".  See an example [here](https://www.elastic.co/guide/en/kibana/current/snapshot-repositories.html#snapshot-repositories-example).  Can I continue calling these examples, or are they tutorials?

**Comment by Titch990** [comment]
@gchaps @szabosteve @alaudazzi @debadair. 

Here's my six-penny worth. While trying to answer Gail's question, I think I may have stumbled on a reasonable definition of the difference between Getting Started, Installation and Configuration, and Tutorials. What do you all think?

TL:DR; Yes, Gail, your example is fine. I think!

I started by trying to answer Gail's question about Examples vs Tutorials. I think our work was less about defining the difference between an Example topic and a Tutorial topic, and more about making sure "Getting started" topics were really "Low-friction, end-to-end paths to doing 'something interesting' with an Elastic product or solution".

At present, in some cases,"Getting Started" is a convenient bucket for something that would better be called "Installation and configuration", which does tell you how to set up stuff, but may not end up doing something interesting (yep, I have some of those, that should be "Installation and Configuration"). In other cases, it's a convenient bucket for a "Tutorial" (which does "something interesting" but may assume you already have done some of the set-up). 

^^^^
So that's my distinction between "Getting started" and "Tutorial". And, because we found serval topics that claimed to be "getting started" which we felt were really tutorials, we felt we needed to make the distinction.

Now, to return to Example vs Tutorial. 

My distinction would be that an example (usually) follows a section of information and is logically included as part of that topic. For example, you describe an API, then give an example of using it. You describe the features you can use to configure a dashboard, then give an example of configuring a dashboard. The user may or may not be able to run that example exactly as written, because they may or may not have set up the other things you refer to in the example, and their environment may not match the environment you describe in the example. But the example should give them enough information to do something similar for themselves in their environment.

Whereas a tutorial is a topic in itself, that pulls together a number of concepts, and may cover several separate aspects of the product. If you follow it as written, you will be able to do something. It won't necessarily be end-to-end, because it is likely to have prerequisites (you need to have installed this, you need to have created this dataset, you need to have run this prior tutorial). But you follow all the instructions, you should be able to complete it as written, whereas you may not be able to follow an example as written (although you may).

And of course, there is a grey area in-between where some examples could be tutorials and vice versa.

Thoughts?

**Comment by szabosteve** [comment]
I close this issue because it is not relevant anymore.
